piv/key.go

```diff
@@ -31,6 +31,10 @@ import (
 	"math/big"
 )
 
+// ErrMismatchingAlgorithms is returned when a cryptographic operation
+// is given keys using different algorithms.
+var ErrMismatchingAlgorithms = errors.New("mismatching key algorithms")
```

Comment:
I wouldn't export this of even use a variable here. Just have the test determine if this returns an error or not.

Reply:
Okay.